src/Libraries/RevitServices/Events/EventHandlerProxy.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+using Autodesk.Revit.DB.Events;
+using Autodesk.Revit.UI.Events;
+using Dynamo.Models;
+
+namespace RevitServices.EventHandler
+{
+    /// <summary>
+    /// This is a event handler proxy class to serve as a proxy between the event publisher and
+    /// the event subscriber
+    /// </summary>
+    public class EventHandlerProxy
```

Comment:
IMHO, 

1) as this class is a proxy for application level event, it should have the same lifetime as the host application, otherwise if it is released somewhere, the subscriber won't be able to notice

2) it is sort of a workaround for Revit event handler memory leak. In the code wherever we use `EventHandlerProxy`, it is actually from `DynamoRevitApplication.EventHandlerProxy`. 

We don't encourage to use `EventHandlerProxy` for all cases except those Revit application events which will cause memory leak, so making it be an inner class makes more sense.